File: VB/GridControlCellTemplate/MainViewModel.vb
Imports DevExpress.Mvvm
Imports DevExpress.Mvvm.DataAnnotations
Imports GridControlCellTemplate.Model

Namespace GridControlCellTemplate

    Public Class MainViewModel
        Inherits ViewModelBase

        Public ReadOnly Property Source As ObservableCollection(Of Record) = New ObservableCollection(Of Record)(Record.GetData(100))

        <Command>
        Public Sub RemoveRow(ByVal item As Record)
            Source.Remove(item)
        End Sub
    End Class
End Namespace

File: CS/GridControlCellTemplate/MainViewModel.cs
using System.Collections.ObjectModel;
using DevExpress.Mvvm;
using DevExpress.Mvvm.DataAnnotations;
using GridControlCellTemplate.Model;

namespace GridControlCellTemplate
{
    public class MainViewModel : ViewModelBase
    {
        public ObservableCollection<Record> Source { get; } = new ObservableCollection<Record>(Record.GetData(100));

        [Command]
        public void RemoveRow(Record item) {
            Source.Remove(item);
        }
    }
}